cadquery/assembly.py

```diff
@@ -106,16 +107,31 @@ def _getAxis(self, arg: Shape) -> Vector:
 
         return rv
 
-    def _getPlane(self, arg: Shape) -> Plane:
+    def _getPln(self, arg: Shape) -> gp_Pln:
 
         if isinstance(arg, Face):
-            normal = arg.normalAt()
+            rv = arg.toPln()
         elif isinstance(arg, (Edge, Wire)):
             normal = arg.normal()
+            origin = arg.Center()
+            plane = Plane(origin, normal=normal)
+            rv = plane.toPln()
         else:
-            raise ValueError(f"Can not get normal from {arg}.")
-        origin = arg.Center()
-        return Plane(origin, normal=normal)
+            raise ValueError(f"Can not construct a plane for {arg}.")
+
+        return rv
+
+    def _getPnt(self, arg: Shape) -> gp_Pnt:
+
+        center = arg.Center()
+        if center.Length > 1e98 and isinstance(arg, Face):
```

Comment:
I think it is better to use `OCP.BRepTools.BRepTools.UVBounds_s` and compare using `OCP.Precision.Precision.IsInfinite_s` (https://dev.opencascade.org/content/how-can-opencascade-display-infinite-plane)

Reply:
Thanks, I was not aware of those functions, that's much better now.